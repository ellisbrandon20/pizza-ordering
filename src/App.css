.App {
  text-align: center;
  background: url(./images/background.jpg) no-repeat center center fixed; 
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
   

  /* ? can't figure out how to dynamically set this height  */
  min-height: 3000px;
}

.background-opacity{
	content: "";
    position: absolute;
    top: 0px;
    right: 0px;
    bottom: 0px;
    left: 0px;
    background-color: rgba(255, 255, 255, 0.4);

	/* ? can't figure out how to dynamically set this height  */
	min-height: 3000px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.transparent-container {
	background: rgba(241, 242, 244, 0.8);
}
