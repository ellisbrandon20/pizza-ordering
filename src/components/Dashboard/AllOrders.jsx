import { Component } from "react";
import { Order } from "./Order";

// TODO - Handle 0 Orders

export class AllOrders extends Component {
	render() {
		const { orders } = this.props;

		return (
			<>
				{orders.map(order => {
					return (
						<Order key={order.Order_ID} order={order} changeUpdateOrders={this.props.changeUpdateOrders} />
					);
				})}
			</>
		);
	}
}