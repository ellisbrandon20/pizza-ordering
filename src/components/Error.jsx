import { Button, Div, Icon, Text } from "atomize";

export const Error = () => {

	return (
		<>
			<Div
				transform='translateY(10rem)'
				w="50rem"
				h="auto"
				m={{ l: 'auto', r: 'auto' }}
				p={{ t: '2rem', b: '2rem' }}
				rounded="lg"
				className="transparent-container"
			>
				<Text tag="h2" textSize="display1">Error: Please Try Again Later</Text>

				<Icon name="AlertSolid" size="5rem" color="danger800" m={{ t: '1rem' }} />
			</Div>
		</>
	);
};