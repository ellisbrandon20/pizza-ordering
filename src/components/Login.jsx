import { Button, Container, Icon, Input, Text } from "atomize";
import { Component } from "react";
import { getLoginToken } from "../apis";
import JWTMgr from "../JWTMgr";

export class Login extends Component {
	constructor(props) {
		super(props);
		this.state = {
			username: "",
			password: ""
		};
	}

	handleSubmit = (e) => {
		e.preventDefault();

		const { username, password } = this.state;
		getLoginToken(username, password)
			.then(responseData => {
				JWTMgr.setToken(responseData.access_token);
				this.props.goToScreen('Dashboard');
			}).catch(_ => {
				this.props.goToScreen('Error');
			});
	}

	handleChange = (input, e) => {
		this.setState({ [input]: e.target.value });
	}

	render() {
		return (
			<>
				<Container
					w="50rem"
					h="auto"
					m={{ l: 'auto', r: 'auto' }}
					p={{ t: '2rem', b: '2rem' }}
					transform='translateY(5rem)'
					rounded="lg"
					className="transparent-container"
				>
					<Text tag="h2" textSize="display1">Login</Text>
					<form onSubmit={this.handleSubmit}>
						<Input
							placeholder="Username"
							value={this.state.username}
							onChange={(e) => this.handleChange('username', e)}
							required
							w="18rem"
							p={{ x: "2.5rem" }}
							m={{ l: 'auto', r: 'auto', t: '0.5rem' }}
							d="block"
							borderColor="danger500"
							focusBorderColor="danger700"
							hoverBorderColor="danger700"
							prefix={
								<Icon
									name="UserSolid"
									color="danger700"
									cursor="pointer"
									pos="relative"
									transform="translate(140%, 15%)"
								/>
							}
						/>
						<Input
							placeholder="Password"
							type="password"
							value={this.state.password}
							onChange={(e) => this.handleChange('password', e)}
							required
							w="18rem"
							p={{ x: "2.5rem" }}
							m={{ l: 'auto', r: 'auto', t: '0.5rem' }}
							d="block"
							borderColor="danger500"
							focusBorderColor="danger700"
							hoverBorderColor="danger700"
							prefix={
								<Icon
									name="LockSolid"
									color="danger700"
									cursor="pointer"
									pos="relative"
									transform="translate(140%, 15%)"
								/>
							}
						/>
						<Button
							type="submit"
							rounded="circle"
							bg="danger700"
							hoverBg="danger600"
							m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
							shadow="2"
							hoverShadow="4"
						>
							Login
						</Button>
					</form>
				</Container>
			</>
		);
	}
}