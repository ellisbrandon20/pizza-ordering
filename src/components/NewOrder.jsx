import { Button, Col, Container, Div, Icon, Row, Text } from "atomize";
import { Component } from "react";
import { createOrder } from "../apis";
import { CrustOption } from "./NewOrderOptions/CrustOption";
import { FlavorOption } from "./NewOrderOptions/FlavorOption";
import { SizeOption } from "./NewOrderOptions/SizeOption";
import { TabelNoOption } from "./NewOrderOptions/TableNoOption";

export class NewOrder extends Component {
	constructor(props) {
		super(props);

		this.state = {
			'newOrders': [{
				'Crust': '',
				'Flavor': '',
				'Size': '',
				'Table_No': undefined,
			}]
		};
	}

	changeSelection = (e, option, index, value) => {
		if (option === 'Table_No') value = parseInt(e.target.value);

		let pizzaOrders = this.state.newOrders;

		pizzaOrders[index][option] = value;

		this.setNewOrders(pizzaOrders);
	};

	handleSubmit = (e) => {
		e.preventDefault();
		const { newOrders } = this.state;

		let successOrder = false;
		newOrders.forEach(order => {
			successOrder = this.createPizzaOrder(order);
		});

		if (successOrder) this.props.goToScreen("SuccessfulOrder");
	}

	createPizzaOrder = (order) => {
		return createOrder(order)
			.then(_ => {
				return true;
			}).catch(_ => {
				this.props.goToScreen('Error');
				return false;
			});
	}

	addPizzaOrder = (e) => {
		e.preventDefault();

		let pizzaOrders = this.state.newOrders;
		pizzaOrders.push({
			'Crust': '',
			'Flavor': '',
			'Size': '',
			'Table_No': undefined,
		});

		this.setNewOrders(pizzaOrders);
	}

	deleteNewOrder = (e, index) => {
		e.preventDefault();

		let pizzaOrders = this.state.newOrders;
		pizzaOrders.splice(index, 1);

		this.setNewOrders(pizzaOrders);
	}

	setNewOrders = (pizzaOrders) => {
		this.setState({
			'newOrders': pizzaOrders
		});
	}

	render() {
		const { newOrders } = this.state;
		return (
			<Div transform='translateY(5rem)'>
				<Text tag="h2" textSize="display1">New Pizza Order</Text>

				<form id='newOrder' onSubmit={this.handleSubmit}>
					{newOrders.map((order, i) => {
						return (
							<Container
								key={i}
								w="50rem"
								h="28rem"
								m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
								rounded="lg"
								className="transparent-container"
							>
								<Text key={`OrderNo-${i}`} tag="h3" textSize="heading">Pizza #{i + 1}</Text>

								<Row>
									<Col size='6'>
										<CrustOption
											key={`crust-${i}`}
											crust={order.Crust}
											index={i}
											changeSelection={this.changeSelection}
										/>
									</Col>
									<Col size='6'>
										<FlavorOption
											key={`flavor-${i}`}
											flavor={order.Flavor}
											index={i}
											changeSelection={this.changeSelection}
										/>
									</Col>
								</Row>
								<Row>
									<Col size='6'>
										<SizeOption
											key={`size-${i}`}
											size={order.Size}
											index={i}
											changeSelection={this.changeSelection}
										/>

									</Col>
									<Col size='6'>
										<TabelNoOption
											key={`tableNo-${i}`}
											tableNo={order.Table_No}
											index={i}
											changeSelection={this.changeSelection}
										/>

									</Col>
								</Row>
								<Row>
									<Col size='12'>
										<Div d="flex">
											<Button
												onClick={(e) => this.addPizzaOrder(e)}
												h="2.5rem"
												w="2.5rem"
												bg="danger700"
												hoverBg="danger600"
												rounded="circle"
												m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
												shadow="2"
												hoverShadow="4"
											>
												<Icon name="Add" size="20px" color="white" />
											</Button>
											<Button
												onClick={(e) => this.deleteNewOrder(e, i)}
												h="2.5rem"
												w="2.5rem"
												bg="danger700"
												hoverBg="danger600"
												rounded="circle"
												m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
												shadow="2"
												hoverShadow="4"
											>
												<Icon name="DeleteSolid" size="20px" color="white" />
											</Button>
										</Div>
									</Col>
								</Row>
							</Container>
						);
					})}

					<Button
						type='submit'
						h="2.5rem"
						w="8rem"
						bg="success800"
						hoverBg="success700"
						rounded="circle"
						m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
						shadow="2"
						hoverShadow="4"
					>
						Place Order
					</Button>
				</form>
			</Div>
		);
	}
}