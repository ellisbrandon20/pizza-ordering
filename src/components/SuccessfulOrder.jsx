import { Button, Div, Icon, Text } from "atomize";

export const SuccessfulOrder = ({ goToScreen }) => {

	return (
		<>
			<Div
				transform='translateY(10rem)'
				w="50rem"
				h="auto"
				m={{ l: 'auto', r: 'auto' }}
				p={{ t: '2rem', b: '2rem' }}
				rounded="lg"
				className="transparent-container"
			>
				<Text tag="h2" textSize="display1">Successfully Submitted Order</Text>

				<Icon name="Checked" size="5rem" color="black" />
				<Button
					onClick={() => goToScreen('Dashboard')}
					prefix={
						<Icon
							name="LongLeft"
							size="16px"
							color="white"
							m={{ r: "0.5rem" }}
						/>
					}
					bg="danger700"
					hoverBg="danger600"
					rounded="circle"
					p={{ r: "1.5rem", l: "1rem" }}
					m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
					shadow="2"
					hoverShadow="4"
				>
					Back To Dashboard
				</Button>
			</Div>
		</>
	);
};