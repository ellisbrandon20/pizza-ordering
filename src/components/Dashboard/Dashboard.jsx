import { Button, Div, Icon, Notification, Text } from "atomize";
import { Component } from "react";
import { getPizza } from "../../apis";
import { AllOrders } from "./AllOrders";
import { Filter } from "./Filter";

export class Dashboard extends Component {
	constructor(props) {
		super(props);

		this.state = {
			'orders': [],
			'filCrust': undefined,
			'filFlavor': undefined,
			'filSize': undefined,
			'filTableNo': undefined,
			'updateOrders': false,
			'successNotif': false
		}
	}

	componentDidMount() {
		this.getPizzaOrders();
	}

	componentDidUpdate(_, prevState) {
		// call API to get up to date orders when an order is deleted
		if (this.state.updateOrders === true) {
			this.getPizzaOrders();
			this.changeUpdateOrders(false);
		}

		// update list of orders based on filter
		let filteredOrders = this.state.orders;
		let filters = {
			'Crust': this.state.filCrust,
			'Flavor': this.state.filFlavor,
			'Size': this.state.filSize,
			'Table_No': this.state.filTableNo ? parseInt(this.state.filTableNo) : undefined
		};

		if (this.didFilterChange(prevState, this.state)) {
			filteredOrders = filteredOrders.map(order => {
				if ((filters.Crust !== undefined && order.Crust !== filters.Crust)
					|| (filters.Flavor !== undefined && order.Flavor !== filters.Flavor)
					|| (filters.Size !== undefined && order.Size !== filters.Size)
					|| (filters.Table_No !== undefined && order.Table_No !== filters.Table_No)
				) {
					return { ...order, 'filtered': false };
				}
				return { ...order, 'filtered': true };
			});

			// must have if statement otherwise this called forever onload/onupdate
			this.setState({ 'orders': filteredOrders });
		}
	}

	didFilterChange = (prevState, state) => {
		// TODO - find better solution
		return (prevState.filCrust !== state.filCrust)
			|| (prevState.filFlavor !== state.filFlavor)
			|| (prevState.filSize !== state.filSize)
			|| (
				(state.filTableNo !== undefined)
				&& (!(isNaN(prevState.filTableNo) && isNaN(state.filTableNo)))
				&& (parseInt(prevState.filTableNo) !== parseInt(state.filTableNo))
			);

	};

	getPizzaOrders = () => {
		// 'filtered' attribute adds ability to show all orders or just the filtered orders without losing the data
		getPizza()
			.then(responseData => {
				const filteredOrders = responseData.map(order => {
					return { ...order, 'filtered': true };
				});

				this.setState({ 'orders': filteredOrders });
			}).catch(_ => {
				this.props.goToScreen('Error');
			});

	}

	changeFilter = (filterCategory, filterValue) => {
		this.setState({
			[filterCategory]: filterValue
		});
	}

	changeUpdateOrders = (msg) => {
		this.setState({
			'updateOrders': msg,
			'successNotif': true
		});
	}

	updateFilter = (filter, value) => {
		this.setState({
			[filter]: value
		});
	}

	render() {
		const filteredOrders = this.state.orders.filter(order => order.filtered === true)

		return (
			<>
				<Notification
					bg="success700"
					isOpen={this.state.successNotif}
					onClose={() => this.setState({ successNotif: false })}
					prefix={
						<Icon
							name="Success"
							color="white"
							m={{ r: "0.5rem" }}
						/>
					}
				>
					Successfully Deleted Pizza
				</Notification>


				<Div m={{ b: '5rem' }} transform='translateY(5rem)'>
					<Div
						w="50rem"
						h="auto"
						m={{ l: 'auto', r: 'auto' }}
						p={{ t: '0.5rem', b: '2rem' }}
						rounded="lg"
						className="transparent-container dashboard-main"
					>
						<Button
							rounded="lg"
							m={{ l: 'auto', r: 'auto', t: '0.5rem', b: '3rem' }}
							bg="danger700"
							hoverBg="danger600"
							onClick={() => this.props.goToScreen('NewOrder')}
						>
							Start New Order
						</Button>

						{/* Current Orders View */}
						<Text tag="h2" textSize="display1" m={{ b: '2rem' }}>
							Current Orders
						</Text>

						<Filter
							filCrust={this.state.filCrust}
							filFlavor={this.state.filFlavor}
							filSize={this.state.filSize}
							filTableNo={this.state.filTableNo}
							updateFilter={this.updateFilter}
						/>
					</Div>

					<AllOrders
						orders={filteredOrders}
						changeUpdateOrders={this.changeUpdateOrders}
					/>
				</Div>
			</>
		);
	}
}