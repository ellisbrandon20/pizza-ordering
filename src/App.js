import { Component } from 'react';
import { Login } from './components/Login';
import { NewOrder } from './components/NewOrder';
import { SuccessfulOrder } from './components/SuccessfulOrder';
import { Dashboard } from './components/Dashboard/Dashboard';
import { Text } from 'atomize';
import { ErrorScreen } from './components/Error';
import './App.css';

class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			currentScreen: 'Login'
		};
	}

	goToScreen = (screen) => {
		this.setState({
			'currentScreen': screen
		});
	}

	render() {
		const { currentScreen } = this.state;
		return (
			<div className="App">
				<div className="background-opacity">

					<Text
						tag="h1"
						textSize="display3"
						m={{ b: '2rem' }}
						transform='translateY(5rem)'
					>
						Jimmy's Pizzeria
					</Text>

					{/* Login Page  */}
					{currentScreen === 'Login' &&
						<Login goToScreen={this.goToScreen} />
					}

					{/* Home Page (Create Order / View Orders) */}
					{currentScreen === 'Dashboard' &&
						<Dashboard goToScreen={this.goToScreen} />
					}

					{/* Create New Order Screen */}
					{currentScreen === 'NewOrder' &&
						<NewOrder goToScreen={this.goToScreen} />
					}

					{/* Submission successful */}
					{currentScreen === 'SuccessfulOrder' &&
						<SuccessfulOrder goToScreen={this.goToScreen} />
					}

					{/* Standard Error page */}
					{currentScreen === 'Error' &&
						<ErrorScreen />
					}
				</div>
			</div>

		);
	}
}

export default App;
