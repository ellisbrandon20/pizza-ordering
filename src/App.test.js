import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Text } from 'atomize';
import { Login } from './components/Login';
import { Dashboard } from './components/Dashboard/Dashboard';
import { NewOrder } from './components/NewOrder';
import { SuccessfulOrder } from './components/SuccessfulOrder';
import { ErrorScreen } from './components/Error';
Enzyme.configure({ adapter: new Adapter() });


test('App Snapshot', () => {
	const wrapper = renderer.create(<App />).toJSON();
	expect(wrapper).toMatchSnapshot();
});

test('should render Text', () => {
	const component = shallow(<App />);
	expect(component.find(Text)).toHaveLength(1);
});

test('should render LogIn Screen', () => {
	const component = shallow(<App />);
	expect(component.find(Login)).toHaveLength(1);
	expect(component.find(Dashboard)).toHaveLength(0);
	expect(component.find(NewOrder)).toHaveLength(0);
	expect(component.find(SuccessfulOrder)).toHaveLength(0);
	expect(component.find(ErrorScreen)).toHaveLength(0);
});

test('should render Dashboard Screen when state is set', () => {
	const component = shallow(<App />);
	component.setState({ 'currentScreen': 'Dashboard' })
	expect(component.find(Dashboard)).toHaveLength(1);
	expect(component.find(Login)).toHaveLength(0);
	expect(component.find(NewOrder)).toHaveLength(0);
	expect(component.find(SuccessfulOrder)).toHaveLength(0);
	expect(component.find(ErrorScreen)).toHaveLength(0);
});

test('should render NewOrder Screen when state is set', () => {
	const component = shallow(<App />);
	component.setState({ 'currentScreen': 'NewOrder' })
	expect(component.find(NewOrder)).toHaveLength(1);
	expect(component.find(Dashboard)).toHaveLength(0);
	expect(component.find(Login)).toHaveLength(0);
	expect(component.find(SuccessfulOrder)).toHaveLength(0);
	expect(component.find(ErrorScreen)).toHaveLength(0);
});

test('should render SuccessfulOrder Screen when state is set', () => {
	const component = shallow(<App />);
	component.setState({ 'currentScreen': 'SuccessfulOrder' })
	expect(component.find(SuccessfulOrder)).toHaveLength(1);
	expect(component.find(Dashboard)).toHaveLength(0);
	expect(component.find(Login)).toHaveLength(0);
	expect(component.find(NewOrder)).toHaveLength(0);
	expect(component.find(ErrorScreen)).toHaveLength(0);
});

test('should render Error Screen when state is set', () => {
	const component = shallow(<App />);
	component.setState({ 'currentScreen': 'Error' })
	expect(component.find(ErrorScreen)).toHaveLength(1);
	expect(component.find(Dashboard)).toHaveLength(0);
	expect(component.find(Login)).toHaveLength(0);
	expect(component.find(NewOrder)).toHaveLength(0);
	expect(component.find(SuccessfulOrder)).toHaveLength(0);
});
