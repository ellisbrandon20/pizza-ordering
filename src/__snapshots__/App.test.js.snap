// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
<div
  className="App"
>
  <div
    className="background-opacity"
  >
    <h1
      className=""
    >
      Jimmy's Pizzeria
    </h1>
    <div
      className="transparent-container "
    >
      <h2
        className=""
      >
        Login
      </h2>
      <form
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <svg
            className=""
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M21,21 L3,21 C3,16.275 6.48532777,14 12,14 C17.5146722,14 21,16.275 21,21 Z M12,12 C9.8,12 8,10.2 8,8 C8,5.8 9.8,4 12,4 C14.2,4 16,5.8 16,8 C16,10.2 14.2,12 12,12 Z"
              fill="#F4541D"
              fillRule="evenodd"
            />
          </svg>
          <input
            className=""
            d="block"
            m={
              Object {
                "l": "auto",
                "r": "auto",
                "t": "0.5rem",
              }
            }
            onChange={[Function]}
            placeholder="Username"
            required={true}
            value=""
          />
        </div>
        <div
          className=""
        >
          <svg
            className=""
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M7,9 L7,8 C7,5.23857625 9.23857625,3 12,3 C14.7614237,3 17,5.23857625 17,8 L17,9 L19,9 C19.5522847,9 20,9.44771525 20,10 L20,20 C20,20.5522847 19.5522847,21 19,21 L5,21 C4.44771525,21 4,20.5522847 4,20 L4,10 C4,9.44771525 4.44771525,9 5,9 L7,9 Z M9,9 L15,9 L15,8 C15,6.34314575 13.6568542,5 12,5 C10.3431458,5 9,6.34314575 9,8 L9,9 Z M12,13 C11.4477153,13 11,13.4477153 11,14 L11,16 C11,16.5522847 11.4477153,17 12,17 C12.5522847,17 13,16.5522847 13,16 L13,14 C13,13.4477153 12.5522847,13 12,13 Z"
              fill="#F4541D"
              fillRule="evenodd"
            />
          </svg>
          <input
            className=""
            d="block"
            m={
              Object {
                "l": "auto",
                "r": "auto",
                "t": "0.5rem",
              }
            }
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className=""
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
